server/py/services/api/tests/unit/db/test_runs.py

```diff
@@ -425,6 +425,46 @@ def test_update_run_success(self):
             assert run["status"]["end_time"] is not None
             assert update_labels_mock.call_count == 0
 
+    def test_store_and_update_run_from_terminal_state_to_non_terminal_state(self):
```

Comment:
I suggest adding to the test that 2 consecutive updates to `state=completed` keeps the `end_time` of the first update and the second update doesn't change the field.